opm/output/eclipse/RestartIO.cpp

```diff
@@ -249,8 +255,9 @@ RestartValue load( const std::string& filename,
 
         if (ecl_file_view_has_kw( file_view , key.c_str())) {
             const ecl_kw_type * ecl_kw = ecl_file_view_iget_named_kw( file_view , key.c_str() , 0 );
-            std::vector<double> data = double_vector( ecl_kw );
-            rst_value.extra[ key ] = std::move(data);
+            const double * data_ptr = ecl_kw_get_double_ptr( ecl_kw );
```

Comment:
There are a few things I did not see:
 - I could not find where the optional keywords are set to be written with `double`? And how are they separated from the non-optional ones at write time (the optionality is only passed in at read time).
 - Should we not read both `float` and `double` interchangably in any case?